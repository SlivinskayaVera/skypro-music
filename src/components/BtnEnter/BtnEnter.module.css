.button {
    width: 278px;
    height: 52px;
    background-color: #580EA2;
    border-radius: 6px;
    margin-left: 4px;
    border: none;
    margin-top: 30px;
    margin-bottom: 20px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;

    font-style: normal;
    font-weight: 400;
    font-size: 18px;
    line-height: 24px;
    letter-spacing: -0.05px;
    color: #FFFFFF;
}

.button:hover {
    background-color: #3F007D;
}

.button:active {
    background-color: #271A58;
}
