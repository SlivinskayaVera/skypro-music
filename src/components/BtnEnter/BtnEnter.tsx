import styles from "./BtnEnter.module.css";

type BtnEnterType = {
  title: string;
  onClick: () => void;
};

export function BtnEnter({ title, onClick }: BtnEnterType) {
  function handleBtnClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
    e.preventDefault();
    onClick();
  }

  return (
    <button onClick={(e) => handleBtnClick(e)} className={styles.button}>
      {title}
    </button>
  );
}
