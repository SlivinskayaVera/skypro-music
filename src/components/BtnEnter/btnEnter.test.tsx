import { BtnEnter } from "./BtnEnter";
import renderer from "react-test-renderer";

describe("Login button", () => {
  test("is the button name displayed correctly", () => {
    const component = renderer.create(<BtnEnter title="React" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
