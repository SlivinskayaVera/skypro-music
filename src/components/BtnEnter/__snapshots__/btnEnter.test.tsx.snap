// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login button is the button name displayed correctly 1`] = `
<button
  className="button"
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    React
  </a>
</button>
`;
